ext/RMagick/extconf.rb

```diff
@@ -201,6 +205,27 @@ def check_partial_imagemagick_versions
        end
     end
 
+    # issue #169
+    # set ARCHFLAGS appropriately for OSX
+    def set_archflags_for_osx
+      archflags = []
+      fullpath = `which convert`
+      fileinfo = `file #{fullpath}`
+
+      # default ARCHFLAGS
+      archs = $ARCH_FLAG.scan(/-arch\s+(\S+)/).flatten
+
+      archs.each do |arch|
+        if fileinfo.include?(arch)
+          archflags << "-arch #{arch}"
+        end
+      end
+
+      if archflags.length != 0
+        $ARCH_FLAG = archflags.join(' ')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.